### PR TITLE
Fix warnings (shadow and conversion)

### PR DESCRIPTION
**Pull request by crtrott**
The extent and stride thing shadow something new in utility.hpp

### CODE REVIEW

File: include/experimental/__p0009_bits/layout_stride.hpp [hunk 1]
```diff
@@ -651,15 +651,15 @@ constexpr void validate_strides(with_rank<N>, Layout, const Extents& ext, const
 
   constexpr auto is_left = std::is_same<Layout, layout_left>::value;
 
-  typename Extents::index_type stride = 1;
+  typename Extents::index_type stride_ = 1;
```
Comment (mhoemmen) [inline include/experimental/__p0009_bits/layout_stride.hpp]: Should we consider avoiding use of trailing underscore for variables that aren't nonpublic data members of the class?

```suggestion
  typename Extents::index_type the_stride = 1;
```

File: include/experimental/__p0009_bits/layout_stride.hpp [hunk 2]
```diff
@@ -651,15 +651,15 @@ constexpr void validate_strides(with_rank<N>, Layout, const Extents& ext, const
...
 
   for (std::size_t r = 0; r < N; r++) {
     const std::size_t s = is_left ? r : N - 1 - r;
 
-    MDSPAN_IMPL_PRECONDITION(common_integral_compare(stride, other.stride(s))
+    MDSPAN_IMPL_PRECONDITION(common_integral_compare(stride_, other.stride(s))
```
Comment (mhoemmen) [inline include/experimental/__p0009_bits/layout_stride.hpp]:
```suggestion
    MDSPAN_IMPL_PRECONDITION(common_integral_compare(the_stride, other.stride(s))
```

File: include/experimental/__p0009_bits/layout_stride.hpp [hunk 3]
```diff
@@ -651,15 +651,15 @@ constexpr void validate_strides(with_rank<N>, Layout, const Extents& ext, const
...
                              and "invalid strides for layout_{left,right}");
 
-    stride *= ext.extent(s);
+    stride_ *= ext.extent(s);
```
Comment (mhoemmen) [inline include/experimental/__p0009_bits/layout_stride.hpp]:
```suggestion
    the_stride *= ext.extent(s);
```

**Review by mhoemmen — APPROVED** [review]
Thanks for the fixes! : - )

File: include/experimental/__p0009_bits/layout_stride.hpp (hunk 3 above)
Comment (crtrott) [reply include/experimental/__p0009_bits/layout_stride.hpp]: renamed to `expected_stride`

**Comment by nmm0** [comment]
I don't think our CI is actually testing Wshadow

**Comment by masterleinad** [comment]
> I don't think our CI is actually testing Wshadow

Looks like the GitHub Workflows don't test with any warning flags and https://github.com/kokkos/mdspan/blob/stable/README.md#running-tests mentions that a lot of the configurations would not build with warnings as errors (depending on the C++ standard).

**Comment by crtrott** [comment]
> I don't think our CI is actually testing Wshadow

I know. But we can touch CI separately.

**Review by nmm0 — APPROVED** [review]
Looks good to me!

**Pull request merged**